test/hexpm/repository/package_test.exs

```diff
@@ -186,12 +186,40 @@ defmodule Hexpm.Repository.PackageTest do
     Hexpm.Repo.refresh_view(Hexpm.Repository.PackageDependant)
 
     assert ["ecto", "phoenix"] =
-             Package.all([repository], 1, 10, "depends:poison", :name, nil)
+             Package.all([repository], 1, 10, "depends:#{repository.name}:poison", :name, nil)
              |> Repo.all()
              |> Enum.map(& &1.name)
 
     assert ["phoenix"] =
-             Package.all([repository], 1, 10, "depends:poison depends:ecto", nil, nil)
+             Package.all(
+               [repository],
+               1,
+               10,
+               "depends:#{repository.name}:poison depends:#{repository.name}:ecto",
+               nil,
+               nil
+             )
+             |> Repo.all()
+             |> Enum.map(& &1.name)
+  end
+
+  test "search dependants is scoped to current repo", %{repository: repository} do
+    private_repo = insert(:repository)
+    insert(:package, name: "nerves", repository_id: repository.id)
+    poison = insert(:package, name: "poison", repository_id: repository.id)
+    ecto = insert(:package, name: "ecto", repository_id: private_repo.id)
+    phoenix = insert(:package, name: "phoenix", repository_id: repository.id)
+
+    rel = insert(:release, package: ecto)
+    insert(:requirement, release: rel, dependency: poison, requirement: "~> 1.0")
+    rel = insert(:release, package: phoenix)
+    insert(:requirement, release: rel, dependency: poison, requirement: "~> 1.0")
+    insert(:requirement, release: rel, dependency: ecto, requirement: "~> 1.0")
+
+    Hexpm.Repo.refresh_view(Hexpm.Repository.PackageDependant)
+
+    assert ["phoenix"] =
+             Package.all([repository], 1, 10, "depends:#{repository.name}:poison", :name, nil)
```

Comment:
@supersimple @ericmj I just realised that in [the original issue](https://github.com/hexpm/hexpm/issues/850#issuecomment-530241707) we said the filter would be `depends:acme/foo`, here we have `depends:acme:foo`. Worth changing it? I don't have a strong opinion either way.

Should we change it?

Reply:
Yes, we should change it. I am looking into how we can refactor the dependants in general.